### PR TITLE
ZipType Parser

### LINKED ISSUE #2: Add ZipType Parser
I imagine it's a common use case to import addresses. I propose adding a zip type parser to the gem that handles leading zeros that get dropped when a CSV is saved (e.g., when a user saves an Excel file as CSV).

### PR DESCRIPTION
**Pull request by inveterateliterate**
Resolves #2 

- Assumes US zip codes (looking for 5 digits or 5-4 digits)
- Returns value as is if it's a zip + 4 (includes dash)
- Returns a 5 character string with zeros in front of the value passed in, if the value came in with fewer than 5 characters.

### CODE REVIEW

File: lib/csv_importable/type_parser/zip_type_parser.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module CSVImportable
+  class TypeParser::ZipTypeParser < TypeParser
+    def parse_val
+      not_digits unless value.count('0-9') == value.length
+      value = '0' + value until value.length >= 5
+      value
```
Comment (inveterateliterate) [inline lib/csv_importable/type_parser/zip_type_parser.rb]: For now, this is specific to the US. This is also assuming that if someone enters a zip + 4 (that has the dash), it appears the CSV does not drop the leading zeros, so we're only worried about zip values that are fewer than 5 characters.

File: lib/csv_importable/type_parser/zip_type_parser.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module CSVImportable
+  class TypeParser::ZipTypeParser < TypeParser
+    def parse_val
+      val = value.delete('-')
+      not_digits unless val.count('0-9') == val.length
```
Comment (inveterateliterate) [inline lib/csv_importable/type_parser/zip_type_parser.rb]: In the US, zip codes do not have any alpha characters. Other countries do though.

File: lib/csv_importable/type_parser/zip_type_parser.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      return value if val.length == 9
```
Comment (inveterateliterate) [inline lib/csv_importable/type_parser/zip_type_parser.rb]: Return the value if it came in as a zip + 4. It appears that the CSV won't drop the leading zeros if there's a dash (treats it as a string I believe).

File: lib/csv_importable/type_parser/zip_type_parser.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      val = '0' + val until val.length >= 5
+      val
```
Comment (inveterateliterate) [inline lib/csv_importable/type_parser/zip_type_parser.rb]: Adds zeros until we get five digits

File: lib/csv_importable/type_parser/zip_type_parser.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    def not_digits
+      raise
+    end
+
+    def error_message
+      "Invalid value for column: #{key}. Value should contain only numbers or a dash."
+    end
```
Comment (inveterateliterate) [inline lib/csv_importable/type_parser/zip_type_parser.rb]: Let me know if this error message is too specific.

**Comment by francirp** [comment]
@inveterateliterate should we clarify by calling this class "USZipTypeParser" perhaps?

**Comment by francirp** [comment]
Otherwise, LGTM!

**Comment by inveterateliterate** [comment]
so @francirp it would be like `pull_us_zip(val)`? I think that's fine!

File: lib/csv_importable/type_parser/us_zip_type_parser.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+module CSVImportable
+  class TypeParser::USZipTypeParser < TypeParser
+    def parse_val
+      val = value.delete('-')
+      not_digits unless val.count('0-9') == val.length
+      return value if val.length == 9
+      val.rjust(5, '0')
```
Comment (inveterateliterate) [inline lib/csv_importable/type_parser/us_zip_type_parser.rb:7]: this literally does: `val = '0' + val until val.length >= 5` :smile:

**Pull request merged**